File: September 2022/September_29.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> (a[1] == b[1]) ? (a[0] - b[0]) : (a[1] - b[1]));
        for(int i : arr)
            pq.add(new int[]{i, Math.abs(x - i)});
        List<Integer> list = new ArrayList<>(k);
        while(k-- > 0)
            list.add(pq.remove()[0]);
        Collections.sort(list);
        return list;
    }
}

File: September 2022/September_29.cpp
class Solution {
public:
    static bool cmp(pair<int, int> &a, pair<int, int> &b){
        if(a.second == b.second)
            return a.first < b.first;
        return a.second < b.second;
    }
    
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        vector<pair<int, int> > vec;
        for(int i : arr)
            vec.push_back({i, abs(x - i)});
        sort(vec.begin(), vec.end(), cmp);
        vector<int> ans(k);
        while(k-- > 0)
            ans[k] = vec[k].first;
        sort(ans.begin(), ans.end());
        return ans;
    }
};
